### PR TITLE
Scan-Build Fixes

### PR DESCRIPTION
**Pull request by ejohnstown**
1. Fixed a couple possible 0 byte allocations.
2. Fixed a couple missed frees due to error conditions.
3. Fixed a possible double free.
4. Fixed a possible garbage value in binary operation.
5. Fix for a possible dereference of a NULL pointer.
6. Fix for a few possible uninitialized pointers.
7. Fix for a possible free of global data.

To recreate:
    $ scan-build ./configure --disable-shared --enable-opensslextra=x509small --disable-memory
    $ scan-build ./configure --disable-shared --enable-opensslextra --disable-memory
    $ scan-build ./configure --enable-sp --enable-sp-asm --enable-sp-math
    $ scan-build ./configure --enable-sp=small --enable-smallstack --enable-smallstackcache CPPFLAGS="-DECC_CACHE_CURVE -DHAVE_WOLF_BIGINT"

(Still one issue outstanding using the second configure option. Leaving out disable memory doesn't report potential errors.)

### CODE REVIEW

**Comment by dgarske** [comment]
@ejohnstown , can you also look into this report. It happening on master.

```
wolfcrypt/src/sp_int.c:990:34: warning: The left operand of '>' is a garbage value due to array index out of bounds
                if (trial->dp[j] > sa->dp[j + o]) {
                    ~~~~~~~~~~~~ ^
1 warning generated.
 Make: Passed
```

File: wolfcrypt/src/sp_c64.c [hunk 1]
```diff
@@ -2843,9 +2843,9 @@ int sp_RsaPublic_2048(const byte* in, word32 inLen, mp_int* em, mp_int* mm,
 {
 #ifdef WOLFSSL_SP_SMALL
     sp_digit* d = NULL;
-    sp_digit* a;
-    sp_digit* m;
-    sp_digit* r;
+    sp_digit* a = NULL;
```
Comment (dgarske) [inline wolfcrypt/src/sp_c64.c:2846]: These changes will need to be made in the scripts repo and when made you will also see the same ones in c32. You can reproduce with `CFLAGS"-DSP_WORD_SIZE=32"`

File: wolfcrypt/src/sp_int.c [hunk 2]
```diff
@@ -2316,7 +2319,7 @@ int sp_prime_is_prime_ex(sp_int* a, int t, int* result, WC_RNG* rng)
 #endif
 
     if (a == NULL || result == NULL || rng == NULL)
-        err = MP_VAL;
+        return MP_VAL;
```
Comment (SparkiDev) [inline wolfcrypt/src/sp_int.c:2322]: Prefer that the next if statement has err check.

Comment (ejohnstown) [reply wolfcrypt/src/sp_int.c:2322]: OK

**Review by dgarske — APPROVED** [review]

**Pull request merged**